openmdao/drivers/tests/test_scipy_optimizer.py

```diff
@@ -1497,6 +1497,301 @@ def test_multiple_objectives_error(self):
+
+        def rastrigin(x):
+            a = 10  # constant
+            return np.sum(np.square(x) - a * np.cos(2 * np.pi * x)) + a * np.size(x)
+
+        class Rastrigin(ExplicitComponent):
+
+            def setup(self):
+                self.add_input('x', 0.5 * np.ones(size))
+                self.add_output('f', 0.5)
+
+            def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+                x = inputs['x']
+                outputs['f'] = rastrigin(x)
+
+        prob = Problem()
+        model = prob.model
+
+        model.add_subsystem('indeps', IndepVarComp('x', np.ones(size)), promotes=['*'])
+        model.add_subsystem('rastrigin', Rastrigin(), promotes=['*'])
+
+        prob.driver = driver = ScipyOptimizeDriver()
+        driver.options['optimizer'] = 'differential_evolution'
+        driver.options['disp'] = False
+        driver.options['tol'] = 1e-9
+
+        model.add_design_var('x', lower=-5.12 * np.ones(size), upper=5.12 * np.ones(size))
+        model.add_objective('f')
+        prob.setup()
+        prob.run_driver()
+        assert_rel_error(self, prob['x'], np.zeros(size), 1e-6)
+        assert_rel_error(self, prob['f'], 0.0, 1e-6)
+
+    def test_differential_evolution_bounded(self):
+        # Source of example:
+        # https://scipy.github.io/devdocs/generated/scipy.optimize.dual_annealing.html
+        # In this example the minimum is not the unbounded global minimum.
+
+        from openmdao.api import Problem, IndepVarComp, ScipyOptimizeDriver
+
+        size = 3  # size of the design variable
+
+        def rastrigin(x):
+            a = 10  # constant
+            return np.sum(np.square(x) - a * np.cos(2 * np.pi * x)) + a * np.size(x)
+
+        class Rastrigin(ExplicitComponent):
+
+            def setup(self):
+                self.add_input('x', 0.5 * np.ones(size))
+                self.add_output('f', 0.5)
+
+            def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+                x = inputs['x']
+                outputs['f'] = rastrigin(x)
+
+        prob = Problem()
+        model = prob.model
+
+        model.add_subsystem('indeps', IndepVarComp('x', np.ones(size)), promotes=['*'])
+        model.add_subsystem('rastrigin', Rastrigin(), promotes=['*'])
+
+        prob.driver = driver = ScipyOptimizeDriver()
+        driver.options['optimizer'] = 'differential_evolution'
+        driver.options['disp'] = False
+        driver.options['tol'] = 1e-9
+
+        model.add_design_var('x', lower=-2.0 * np.ones(size), upper=-0.5 * np.ones(size))
+        model.add_objective('f')
+        prob.setup()
+        prob.run_driver()
+        assert_rel_error(self, prob['x'], -np.ones(size), 1e-2)
+        assert_rel_error(self, prob['f'], 3.0, 1e-2)
+
+    @unittest.skipUnless(LooseVersion(scipy_version) >= LooseVersion("1.2"),
+                         "scipy >= 1.2 is required.")
+    def test_shgo(self):
```

Comment:
Péter,

After adding a scipy 1.2 configuration to our Travis CI, we have found that this test does not pass (I was able to get it to pass locally by removing the bounds on the design variable).  We would appreciate it if you could take a look at your convenience.

Thanks!

Reply:
Also see Issue #909

Reply:
Hmm, for me the test passes with SciPy 1.2.1 on Windows, I will try to figure it out.